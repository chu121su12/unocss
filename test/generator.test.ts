import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'
import presetMini from '@unocss/preset-mini'

describe('unocss generator', () => {
  test('generator can enable noMerge option', async () => {
    const uno = createGenerator({
      presets: [
        presetMini(),
      ],
    })

    const tokens = [
      'left-0',
      'active:left-0',
      'focus:left-0',
      'hover:left-0',
      'active:focus:left-0',
      'group-focus:left-0',
      'group-hover:left-0',
    ].join(' ')

    const { css: css1 } = await uno.generate(tokens, { preflights: false, noMerge: false })
    expect(css1).toMatchSnapshot()
    const { css: css2 } = await uno.generate(tokens, { preflights: false, noMerge: true })
    expect(css2).toMatchSnapshot()
  })
})
