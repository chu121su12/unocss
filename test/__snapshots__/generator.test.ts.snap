// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unocss generator > generator can enable noMerge option 1`] = `
"/* layer: default */
.group:hover .group-hover\\\\:left-0,
.hover\\\\:left-0:hover,
.left-0{left:0;}
.focus\\\\:left-0:focus{left:0;}
.active\\\\:focus\\\\:left-0:focus:active{left:0;}
.active\\\\:left-0:active{left:0;}
.group:focus .group-focus\\\\:left-0{left:0;}"
`;

exports[`unocss generator > generator can enable noMerge option 2`] = `
"/* layer: default */
.left-0{left:0;}
.hover\\\\:left-0:hover{left:0;}
.focus\\\\:left-0:focus{left:0;}
.active\\\\:focus\\\\:left-0:focus:active{left:0;}
.active\\\\:left-0:active{left:0;}
.group:hover .group-hover\\\\:left-0{left:0;}
.group:focus .group-focus\\\\:left-0{left:0;}"
`;
